broker-core/src/test/java/io/zeebe/broker/workflow/UpdatePayloadTest.java

```diff
@@ -1,300 +0,0 @@
-  private static final BpmnModelInstance WORKFLOW =
-      Bpmn.createExecutableProcess("process")
-          .startEvent()
-          .serviceTask(
-              "task-1",
-              t ->
-                  t.zeebeTaskType("task-1")
-                      .zeebeTaskRetries(5)
-                      .zeebeOutput("$.jsonObject", "$.obj"))
-          .serviceTask("task-2", t -> t.zeebeTaskType("task-2").zeebeTaskRetries(5))
-          .endEvent()
-          .done();
-
-  public EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
-  public ClientApiRule apiRule = new ClientApiRule(brokerRule::getClientAddress);
-
-  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(apiRule);
-
-  private PartitionTestClient testClient;
-
-  @Before
-  public void init() {
-    testClient = apiRule.partitionClient();
-  }
-
-  @Test
-  public void shouldUpdatePayload() {
-    // given
-    testClient.deploy(WORKFLOW);
-
-    final long workflowInstanceKey = testClient.createWorkflowInstance("process");
-
-    final Record<WorkflowInstanceRecordValue> activityInstanceEvent =
-        waitForActivityActivatedEvent();
-
-    // when
-    final ExecuteCommandResponse response =
-        updatePayload(
-            activityInstanceEvent.getKey(), MsgPackUtil.asMsgPackReturnArray("{'foo':'bar'}"));
-
-    // then
-    assertThat(response.getIntent()).isEqualTo(WorkflowInstanceIntent.PAYLOAD_UPDATED);
-
-    final Record<WorkflowInstanceRecordValue> updateCommand =
-        testClient.receiveFirstWorkflowInstanceCommand(WorkflowInstanceIntent.UPDATE_PAYLOAD);
-    final Record<WorkflowInstanceRecordValue> updatedEvent =
-        testClient.receiveFirstWorkflowInstanceEvent(WorkflowInstanceIntent.PAYLOAD_UPDATED);
-
-    assertThat(updatedEvent.getSourceRecordPosition()).isEqualTo(updateCommand.getPosition());
-    assertThat(updatedEvent.getKey()).isEqualTo(activityInstanceEvent.getKey());
-    assertThat(updatedEvent.getValue().getWorkflowInstanceKey()).isEqualTo(workflowInstanceKey);
-
-    assertWorkflowInstancePayload(updatedEvent, "{'foo':'bar'}");
-  }
-
-  @Test
-  public void shouldUpdateWithNilPayload() {
-    // given
-    testClient.deploy(WORKFLOW);
-
-    testClient.createWorkflowInstance("process");
-
-    final Record<WorkflowInstanceRecordValue> activityInstanceEvent =
-        waitForActivityActivatedEvent();
-
-    // when
-    final ExecuteCommandResponse response =
-        updatePayload(activityInstanceEvent.getKey(), MsgPackHelper.NIL);
-
-    // then
-    assertThat(response.getIntent()).isEqualTo(WorkflowInstanceIntent.PAYLOAD_UPDATED);
-    final Record<WorkflowInstanceRecordValue> updatedEvent =
-        testClient.receiveFirstWorkflowInstanceEvent(WorkflowInstanceIntent.PAYLOAD_UPDATED);
-
-    assertWorkflowInstancePayload(updatedEvent, "{}");
-  }
-
-  @Test
-  public void shouldUpdateWithZeroLengthPayload() {
-    // given
-    testClient.deploy(WORKFLOW);
-
-    testClient.createWorkflowInstance("process");
```

Comment:
Do we have replacement for theses tests?

Reply:
Yes, but more indirectly; some are tested in the `UpdateVariableDocument*` integration/unit tests, some are tested in the variable state itself (e.g. the propagation logic for example, or handling of an empty payload).

We can discuss f2f if and which we should keep as IT if you want

Reply:
Didn't found tests for the following scenarios:

 * shouldUpdateWithNilPayload
 * shouldUpdateWithZeroLengthPayload
 * shouldUpdatePayloadWhenActivityActivated
 * shouldUpdatePayloadWhenCatchEventIsEntered
 * shouldThrowExceptionForInvalidPayload
 * shouldRejectUpdateForNonExistingWorkflowInstance
 * shouldRejectUpdateForCompletedWorkflowInstance

Reply:
I think most of them are tested in the [QA tests](https://github.com/zeebe-io/zeebe/blob/e700452c667c8617250f77141a5829c1172b2bf1/qa/integration-tests/src/test/java/io/zeebe/broker/it/workflow/UpdatePayloadTest.java)

Reply:
A thanks searched for UpdateVariable not payload :D Then `shouldUpdateWithZeroLengthPayload` ("{}" as payload ?) 

> t's not really relevant anymore since the new lifecycle means it's the same as the activity activated,

Is this really the case? There still different processors involved or not?

> I'm not really sure `shouldRejectUpdateForCompletedWorkflowInstance` is useful, for example, since `UpdateVariableDocumentProcessorTest` tests when an element does not exist, which is basically the same).

Yeah, but here you test end-to-end so you also see how it looks like in the client and that this works as expected.